libethash-cuda/ethash_cuda_miner_kernel.cu

```diff
@@ -38,7 +38,6 @@ ethash_search(
 	if (cuda_swab64(hash) > d_target) return;
 	uint32_t index = atomicInc(const_cast<uint32_t*>(g_output), SEARCH_RESULT_BUFFER_SIZE - 1) + 1;
 	g_output[index] = gid;
-	__threadfence_system();
```

Comment:
I'll leave it up to @RoBiK75 to comment on the neccesity of this, as he has submitted this change.